### DISCUSSION TITLE
Strange problem in Flutter/Firebase

### DISCUSSION THREAD

**Discussion by harry-dickson** [square/okio#1114] [General]
Does this trace mean anything to anyone here?
```
W/Firestore( 5850): (24.1.1) [WatchStream]: (150415a) Stream closed with status: Status{code=INTERNAL, description=error in frame handler, cause=java.lang.NoSuchMethodError: No interface method getBuffer()Lokio/Buffer; in class Lokio/BufferedSource; or its super classes (declaration of 'okio.BufferedSource' appears in /data/app/~~562CMs3V8mJAW5W-e8eBRQ==/care.mpm.app.myguide-vSZFTwtxrkRSO-SPz3Pd5g==/base.apk!classes5.dex)
W/Firestore( 5850): 	at io.grpc.okhttp.OkHttpClientTransport$ClientFrameHandler.data(OkHttpClientTransport.java:1144)
W/Firestore( 5850): 	at io.grpc.okhttp.internal.framed.Http2$Reader.readData(Http2.java:234)
W/Firestore( 5850): 	at io.grpc.okhttp.internal.framed.Http2$Reader.nextFrame(Http2.java:147)
W/Firestore( 5850): 	at io.grpc.okhttp.OkHttpClientTransport$ClientFrameHandler.run(OkHttpClientTransport.java:1103)
W/Firestore( 5850): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/Firestore( 5850): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/Firestore( 5850): 	at java.lang.Thread.run(Thread.java:923)
```

This is a a problem arising some distance downstream of this project. https://github.com/firebase/flutterfire/issues/8501

**Comment by swankjesse** [comment]
Whenever you see NoSuchMethodError, it's almost always related to executing with a different version than you compiled against.

In this case it looks like the runtime Okio version is earlier than 1.16.0 and grpc-okhttp needs that or newer.

https://github.com/square/okio/blob/master/CHANGELOG.md#version-1160

**Reply by harry-dickson** [reply]
Thanks!

**Reply by mujahid-mohammed** [reply]
> Thanks!

So can you please tell me how you fix it?

**Reply by leva0212** [reply]
add the next dependenci line to the build.gradle file in app folder
implementation("com.squareup.okhttp3:okhttp:4.9.3") 
this will fix the problem, then the application will by able to conect to firebase correctly

**Reply by iHamza7** [reply]
@leva0212 thank you very much this worked for me i was trying for 3 days thank you so much

**Comment by swankjesse** [comment]
However you're pulling in a dependency on Okio, make sure it's at least 1.16.0. In regular Android projects this is in `build.gradle`; I don't know anything about Flutter or Firebase.
